### ISSUE TITLE
OSX after 6a0572e

### ISSUE THREAD

**Issue by luzpaz** [nitramr/indigoDock#23]
After the 6a0572e1acfb7e1238767ac1fc35a255d0dd6122 commit OSX isnt displaying the sidebar icons
![screenshot_2015-12-12_05_21_27](https://cloud.githubusercontent.com/assets/4140247/11761659/776fa2e0-a090-11e5-8af9-823bd64ce365.jpg)

Also, how does one switch between dark and light themes ? Can it be done via the UI or does it need to be done via code?

**Comment by nitramr** [comment]
The switch is controlled via theme file. /data/qss/dark.qss

If you change "dark" to "light" you have no icon inversion
Palette::dark {...

The default is the dark theme. Can you check your Debug folder if the folder structure there?
- icons folder and data folder

If the files only in project folder and not in debug folder you don't see the icons.

**Comment by luzpaz** [comment]
```
🍎 Test:indigoDock sphenoid$ pwd
/Users/sphenoid/Develop/indigoDock
🍎 Test:indigoDock sphenoid$ tree
.
├── LICENSE
├── README.md
├── README.md~
├── TestLayout.pro
├── TestLayout.pro.user
├── anglepicker.cpp
├── anglepicker.h
├── colorswatch.cpp
├── colorswatch.h
├── data
│   ├── navigation.json
│   └── qss
│       ├── dark.qss
│       └── dark.qss~
├── icons
│   ├── close.png
│   ├── new.png
│   ├── open.png
│   ├── pan-image.png
│   ├── pan-page.png
│   ├── pan-text.png
│   ├── placeholder.png
│   └── quit.png
├── icons.qrc
├── indigodock.cpp
├── indigodock.h
├── indigodropzone.cpp
├── indigodropzone.h
├── indigoexpandergroup.cpp
├── indigoexpandergroup.h
├── indigomenubar.cpp
├── indigomenubar.h
├── indigopanel.cpp
├── indigopanel.h
├── indigotabbar.cpp
├── indigotabbar.h
├── main.cpp
├── mainwindow.cpp
├── mainwindow.h
├── mainwindow.ui
├── stylefactory.cpp
└── stylefactory.h
```

**Comment by luzpaz** [comment]
I don't see a debug folder

**Comment by nitramr** [comment]
It's not in the structure here... This folder will generate by QT while compiling... you have the option to define a default folder for debug and release output...

If you compile the project in debug mode you have to add the icons and data folder to these folder, because your "debug"-app will start from there.

**Comment by luzpaz** [comment]
Oh, I see. Yea, i'm not compiling indigoDock via debug. I will try and see.

**Comment by luzpaz** [comment]
@nitramr I'm not finding where the debug folder you speak of is being generated during compile time. 

BTW I just pulled down you latest update 1e484ac1101451c0a6a95a8e4d1d2b51dc9c439b to test the theme switcher 
I've run debug and when I try to change the theme i get this in the output:
`Unable to set stylesheet, file not found`
I guess this has to do with https://github.com/nitramr/indigoDock/issues/23#issuecomment-164162031 ?

**Comment by nitramr** [comment]
So, try again... I have modified the project file with predefined output paths. Now the compiling output is within the GitHub project folder (built)... if it not work try following:
- find the place where the program will compiled (in my case a debug folder)
- if you have found the location copy the whole "data" folder to this location...
- now it should work

I'm not sure how if it works on Mac, but I think it's similar like Linux.

PS: yes, the error is related to missing files... in gerneral, all files have to be relative to the application.

**Comment by aoloe** [comment]
at some time it might be better to switch to cmake... this is also what scribus uses... but it does not hurry: as long as qmake works good enough, it's ok to use it :-)

**Comment by luzpaz** [comment]
I found that QtCreator5.6 was creating a temporary folder in a parent directory over the folder I'm compiling indigoDock in. 

The folder is named: `build-TestLayout-Desktop_Qt_5_5_1_clang_64bit-Release/`

I did what you said and copied the `data/` directory from the latest revision in to 
`../build-TestLayout-Desktop_Qt_5_5_1_clang_64bit-Release/`
Still no joy :(
I get a bunch of warnings though: 

```
/Users/sphenoid/Develop/indigoDock copy/configmanager.h:1: warning: 'CONFIGMANAGER_H' is used as a header guard here, followed by #define of a different macro [-Wheader-guard]
#ifndef CONFIGMANAGER_H
        ^~~~~~~~~~~~~~~
```

I don't have time ATM to copy them all, but most of the important ones looked like the above

**Comment by luzpaz** [comment]
please advise

**Comment by nitramr** [comment]
Hi, please try the latest commit. I have also updated the description how to run the IndigoDock tester.
See in readme file step 4.
In short words: After opening the project you have to deactivate the standard out put path for compiling in Qt itself. Instead of the standard path it will used the predefined path in the project file.

If it doesn't work, search for the compiled out put file on your system and copy the "data" folder on same location as your compiled file. You can find the "data" folder inside of the "build" folder from the github project.

Note: this commit is not stable and have some known bugs that I have to fix (see readme too).

Let me know if you need help. ;)

**Comment by luzpaz** [comment]
hey man, is there a way we can automate this whole data/ folder move ? For some reason, it still isn't working for me on OSX. If not that's cool. Keep working and we'll figure it out. I don't want to take you out of the momentum. Exciting stuff! Cheers

**Comment by aoloe** [comment]
hey, ... personally, at some time i would move to cmake (instead of qmake)... then the normal workflow that people are used in scribus can also be used for testing indigodock...
(all in all, qmake is not very well supported nor documented... and most projects stop using it after the starting phase...)

**Comment by luzpaz** [comment]
Issues fixed in 1c32d5eb57b067a407b4cc5f58fc9f5b4a6683df and 3330888137c2109d7607d717f57ec0100483e04c
Thanks @nitramr! 
Closing
